From 11c02a8414b6510f1b2c8706cc627cb0342b5254 Mon Sep 17 00:00:00 2001
From: zengjing <zengjing@xiaomi.com>
Date: Sat, 29 Sep 2018 13:19:15 +0800
Subject: [PATCH 37/43] Visualizer: fix native crash when visualizer release

Visualizer should not be destroyed before mCaptureThread exit.

Test: CTS VisualizerTest
Change-Id: I0cda2ff2be8ce487e10b5311fbdc043fa57d91d4
Signed-off-by: zengjing <zengjing@xiaomi.com>
---
 media/jni/audioeffect/android_media_Visualizer.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/media/jni/audioeffect/android_media_Visualizer.cpp b/media/jni/audioeffect/android_media_Visualizer.cpp
index b7d7b033937..45de36ed3e4 100644
--- a/media/jni/audioeffect/android_media_Visualizer.cpp
+++ b/media/jni/audioeffect/android_media_Visualizer.cpp
@@ -440,6 +440,7 @@ static void android_media_visualizer_native_release(JNIEnv *env,  jobject thiz)
         if (lpVisualizer == 0) {
             return;
         }
+        lpVisualizer->release();
     }
     // delete the JNI data
     VisualizerJniStorage* lpJniStorage =
-- 
2.17.1

